src/state/translations-mixin.ts

```diff
@@ -78,33 +100,41 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
       super.panelUrlChanged(newPanelUrl);
       // this may be triggered before hassConnected
       this._loadFragmentTranslations(
-        this.hass ? this.hass.language : getLocalLanguage(),
+        this.hass ? this.hass.language.language : getLocalLanguage(),
         newPanelUrl
       );
     }
 
-    private _selectLanguage(language: string, saveToBackend: boolean) {
+    private _selectLanguage(
+      language: FrontendTranslationData,
+      saveToBackend: boolean
+    ) {
       if (!this.hass) {
         // should not happen, do it to avoid use this.hass!
         return;
       }
 
       // update selectedLanguage so that it can be saved to local storage
-      this._updateHass({ language, selectedLanguage: language });
+      this._updateHass({
+        language,
+        selectedLanguage: language.language,
```

Comment:
So we now only save `language.language` to localstorage, I think we should store the entire `language` object, so all language settings can be overruled on device level?

Reply:
Done!